### PR TITLE
Updated to use the new map upgrade program.

### PR DESCRIPTION
**Pull request by dsimmons87**
Applies to all OpenRA revisions from end of may 2018. As it often asks
for manual fixes to maps, any advanced maps now just display the console
output so users that are upgrading can see if this is needed.

### CODE REVIEW

**Comment by pchote** [comment]
Can you please register the `MapUpgradeLog`s in the admin interface, with a custom `MapUpgradeLogAdmin` model (see #332) so that they display usefully in the admin list?

**Comment by dsimmons87** [comment]
Ok. Will need to rebase onto your latest changes and do the same thing for these admin logs. Easy enough, but I'm so busy at the moment, it might be a week or two before I can get time. When I do, I'll get this PR added too as I've asked around loads and nobody has time to check it.

File: openra/utility.py [hunk 1]
```diff
@@ -11,14 +11,20 @@
...
 def map_upgrade(mapObject, engine, parser=list(reversed(list(settings.OPENRA_VERSIONS.values())))[0], new_rev_on_upgrade=True, upgrade_if_hash_matches=False, upgrade_if_lint_fails=False):
 
     parser_to_db = parser
+    parser_date = parser.split('-')[1]
     parser = os.path.join(settings.OPENRA_ROOT_PATH, parser)
+    engine_date = engine.split('-')[1]
+
+    # If parser date is newer than end of may 2018 use update-map and the full release name, otherwise use upgrade-map and the date only
+    # due to the syntax for this command changing in OpenRA
+    updated_openra_utility = int(parser_date) > 20180528
```
Comment (pchote) [inline openra/utility.py]: The new commands have an additional constraint: `engine >= release-20171014`.  If someone wants to update from e.g. `release-20160508` to (whatever our next release ends up being) then they will first need to update to `release-20171014` using the the `release-20171014` engine and the old utility, and then to the latest version using the latest engine.

Would inserting something like the following here work?
```python
# HACK: we really shouldn't be assuming engine versions of the form <prefix>-<date>
if engine.split('-')[1] < '20171014':
    mapObject = map_upgrade(mapObject, engine, settings.OPENRA_VERSIONS['release-20171014'], new_rev_on_upgrade, upgrade_if_hash_matches, upgrade_if_lint_fails)
```

The alternative, which i've suggested elsewhere, is to delete all maps and parsers older than `release-20171014` and then support only the new-style updating.

Comment (pchote) [reply openra/utility.py]: #337 removed the web-facing update feature for parsers < 20171014.  I plan to rebase, test, and merge this PR on top of these changes in the next day or two.

**Comment by pchote** [comment]
Rebased and updated to fix my review comments.

**Pull request merged**